persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -56,60 +50,56 @@ public class BalProjectUtils {
-                    for (Diagnostic d : diagnosticResult.errors()) {
-                        errorMessage.append(System.lineSeparator());
-                        errorMessage.append(d.toString());
+            BuildProject buildProject = BuildProject.load(dirPath.toAbsolutePath());;
+            Package currentPackage = buildProject.currentPackage();
+            PackageCompilation compilation = currentPackage.getCompilation();
+            DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+            if (diagnosticResult.hasErrors()) {
+                StringBuilder errorMessage = new StringBuilder();
+                int count = 0;
+                errorMessage.append("Error occurred when validating the project. ");
+                for (Diagnostic d : diagnosticResult.errors()) {
+                    if (d.toString().contains("redeclared symbol")) {
+                        continue;
                     }
+                    errorMessage.append(System.lineSeparator());
+                    errorMessage.append(d.toString());
+                    count += 1;
+                }
+                if (count > 0) {
                     throw new BalException(errorMessage.toString());
                 }
             }
-            try (Stream<Path> walk = Files.walk(dirPath)) {
-                if (walk != null) {
-                    fileList = walk.filter((filePath) -> Files.isRegularFile(filePath) &&
-                                    !filePath.toAbsolutePath().equals(clientEntitiesPath)
-                                    && filePath.normalize().toString().endsWith(EXTENSION_BAL))
-                            .collect(Collectors.toList());
-                    for (Path filePath : fileList) {
-                        EntityMetaData retEntityMetaData = BalSyntaxTreeGenerator.getEntityRecord(filePath);
-                        ArrayList<Entity> retData = retEntityMetaData.entityArray;
-                        ArrayList<ModuleMemberDeclarationNode> retMembers = retEntityMetaData.moduleMembersArray;
-                        if (retData.size() != 0) {
-                            returnMetaData.addAll(retData);
-                            returnModuleMembers.addAll(retMembers);
-                        }
+            ArrayList<String> entityNames = new ArrayList<>();
+            for (Module module : buildProject.currentPackage().modules()) {
+                for (DocumentId documentId : module.documentIds()) {
+                    if (documentId.moduleId().moduleName().trim().endsWith(".clients")) {
+                        continue;
+                    }
+                    Document document = module.document(documentId);
+                    EntityMetaData retEntityMetaData = BalSyntaxTreeGenerator
+                            .getEntityRecord(document.syntaxTree());
+                    ArrayList<Entity> retData = retEntityMetaData.entityArray;
```

Comment:
```suggestion
                    ArrayList<Entity> entityMetaDataList = retEntityMetaData.entityArray;
```

Reply:
updated in [c255915](https://github.com/ballerina-platform/persist-tools/pull/70/commits/c255915e8c07c686564fb7b9c6375f2af3507865)